71_RayTracingPipeline/app_resources/raytrace_shadow.rahit.hlsl

```diff
@@ -0,0 +1,14 @@
+#include "common.hlsl"
+
+[[vk::push_constant]] SPushConstants pc;
+
+[shader("anyhit")]
+void main(inout OcclusionPayload payload, in BuiltInTriangleIntersectionAttributes attribs)
+{
+    const int instID = InstanceID();
+    const STriangleGeomInfo geom = vk::RawBufferLoad < STriangleGeomInfo > (pc.triangleGeomInfoBuffer + instID * sizeof(STriangleGeomInfo));
+    const Material material = nbl::hlsl::_static_cast<Material>(geom.material);
+    
+    payload.attenuation = material.alpha * payload.attenuation;
```

Comment:
alpha 1 means full opacity, not full transparency, did you misname your `material` field?

Reply:
or you have a logic flaw, seeing your material you have `isTransparent() return alpha<1.f` so its definitely a logic error here, you should be multiplying with `(1-alpha)`

Reply:
fixed by me